lib/modules/showImages.js

```diff
@@ -1458,13 +1497,30 @@ function videoAdvanced(options) {
 		time,
 	} = options;
 
-	const element = $(videoAdvancedTemplate(options))[0];
+	// Poster is unnecessary, and will flash if loaded before the video is ready
+	if (autoplay) delete options.poster;
+
+	const element = document.createElement('div');
+	const player = $(videoAdvancedTemplate(options))[0];
+	element.appendChild(player);
 
-	const vid = element.querySelector('video');
-	const sourceElements = vid.querySelectorAll('source');
+	const vid = player.querySelector('video');
 
 	const msgError = element.querySelector('.video-advanced-error');
 
+	const sourceElements = $(options.sources.map(v => {
```

Comment:
Should probably `_.compact` this, because all sources could be `null` (jQuery appears to preserve `null`).
Also `lastSource.addEventListener` below will throw if the last source is null.